### PR TITLE
DOM

### PR DESCRIPTION
**Pull request by OlStani**
### DOM

[Demo](https://olstani.github.io/star-wars-intro/) |
[Code base](https://github.com/OlStani/star-wars-intro)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

### CODE REVIEW

File: submissions/OlStaniev/DOM/script/script.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+for (let i = 0; i < episodes.length; i++) {
+    const navListItem = 
+    `<li class="listItem">
+        <button class="nav__list_item" data-id="${episodes[i].id}">
+            <span class="listTitle">${episodes[i].title}</span>
+            <span class="listSubTitle">${episodes[i].subTitle}</span>
+        </button>
+    </li>`
+    menuList.insertAdjacentHTML('beforeend',navListItem)
+}
```
Comment (madmaxWMFU) [inline submissions/OlStaniev/DOM/script/script.js:42]: To begin with, let's make a load function so that it is clear from where we will read the code.

File: submissions/OlStaniev/DOM/script/script.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    const currentEpisode = episodes.find((episodeNumber) => episodeId === episodeNumber.id)
+    const storyText = `<div class="introText_wrapper">${currentEpisode.title + currentEpisode.subTitle + currentEpisode.text}</div>`
+    introText.innerHTML = storyText
+}
+
+for (let i = 0; i < episodes.length; i++) {
+    const navListItem = 
+    `<li class="listItem">
+        <button class="nav__list_item" data-id="${episodes[i].id}">
+            <span class="listTitle">${episodes[i].title}</span>
```
Comment (madmaxWMFU) [inline submissions/OlStaniev/DOM/script/script.js:37]: Don`t forget about the structure of the menu `nav>ul>li>a` because every episode in the menus is a link.

File: submissions/OlStaniev/DOM/script/script.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    }
+})
+
+function changeText(episodeId) {
+    const currentEpisode = episodes.find((episodeNumber) => episodeId === episodeNumber.id)
+    const storyText = `<div class="introText_wrapper">${currentEpisode.title + currentEpisode.subTitle + currentEpisode.text}</div>`
+    introText.innerHTML = storyText
+}
+
+for (let i = 0; i < episodes.length; i++) {
```
Comment (madmaxWMFU) [inline submissions/OlStaniev/DOM/script/script.js:33]: Consider writing a separate menu rendering function.

Comment (madmaxWMFU) [inline submissions/OlStaniev/DOM/script/script.js:33]: Also, you can try some iteration methods on the [Array](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach)

File: submissions/OlStaniev/DOM/index.html [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        <img src="./img/star-wars.svg" alt="star wars logo" class="header__logo">
+        <button class="header__burgerMenu"></button>
+    </header>
+    <main>
+        <nav class="nav">
+            <ul class="nav__list">
+                <!-- Test text -->
+                <!-- <li class="listItem">
+                    <button class="nav__list_item" data-id="ep4">
+                        <span class="listTitle">Episode IV</span>
```
Comment (madmaxWMFU) [inline submissions/OlStaniev/DOM/index.html:26]: Leave comments in the test or debug thread. In the final code version, they are not needed or used very rarely.

**Review by madmaxWMFU — CHANGES_REQUESTED** [review]
Hello! Great job. There are a couple of things that need to be fixed.

File: submissions/OlStaniev/DOM/script/script.js (hunk 1 above)
Comment (OlStani) [reply submissions/OlStaniev/DOM/script/script.js:42]: я правильно зрозумів? мені потрібно цикл замінити на метод перебору массива та помістити його у функцію? не розумію що маю зробити у другому пункті -" Dont forget about the structure of the menu nav>ul>li>a` because every episode in the menus is a link."

Comment (OlStani) [reply submissions/OlStaniev/DOM/script/script.js:42]: > To begin with, let's make a load function so that it is clear from where we will read the code.

я правильно зрозумів? мені потрібно цикл замінити на метод перебору массива та помістити його у функцію? не розумію що маю зробити у другому пункті -" Dont forget about the structure of the menu nav>ul>li>a` because every episode in the menus is a link."

File: submissions/OlStaniev/DOM/script/script.js (hunk 2 above)
Comment (OlStani) [reply submissions/OlStaniev/DOM/script/script.js]: Це не як навігація по сайту, це типу як табуляція між елементами, тому використав баттон.

Comment (madmaxWMFU) [reply submissions/OlStaniev/DOM/script/script.js]: Okay, then realize some actions: active, hover, focus.

File: submissions/OlStaniev/DOM/script/script.js (hunk 1 above)
Comment (madmaxWMFU) [reply submissions/OlStaniev/DOM/script/script.js:45]: I mean write a function in which we call function, declare handlers, and others.
For example: 
```javascript
const init = () => {
   .....
   renderMenuList();
   menu.addEventListener('click', getEpisode);
   .....
}
```

File: submissions/OlStaniev/DOM/script/script.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
+import episodes from './text-episode.js';
+
+const menuList = document.querySelector('.nav__list');
+const introText = document.querySelector('.introText')
+const burgerIcon = document.querySelector('.header__burgerMenu')
+
+document.addEventListener('click', (event) => {
+    const episodeName = event.target.closest('.nav__list_item')
```
Comment (madmaxWMFU) [inline submissions/OlStaniev/DOM/script/script.js:8]: Why did you add a listener for a document?

File: submissions/OlStaniev/DOM/script/script.js [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    const currentEpisode = episodes.find((episodeNumber) => episodeId === episodeNumber.id)
+    const storyText = `<div class="introText_wrapper"><h2>${currentEpisode.title}</h2><h3>${currentEpisode.subTitle}</h3>${currentEpisode.text}</div>`
+    return storyText
+}
+
+const navList = episodes.reduce((list, episode) =>{
+    return list +
+    `<li class="listItem">
+        <button class="nav__list_item" data-id="${episode.id}">
+            <span class="listTitle">${episode.title}</span>
```
Comment (madmaxWMFU) [inline submissions/OlStaniev/DOM/script/script.js:37]: You can try [destructuring assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment).

File: submissions/OlStaniev/DOM/script/text-episode.js [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+    of trade routes to outlying star
+    systems is in dispute.</p>
+    <p>Hoping to resolve the matter
+    with a blockade of deadly
+    battleships, the greedy Trade
+    Federation has stopped all
+    shipping to the small planet
+    of Naboo.</p>
+    <p>While the Congress of the
+    Republic endlessly debates
```
Comment (madmaxWMFU) [inline submissions/OlStaniev/DOM/script/text-episode.js:16]: You can also get rid of the rest of the tags

**Review by madmaxWMFU — CHANGES_REQUESTED** [review]

File: submissions/OlStaniev/DOM/script/script.js (hunk 5 above)
Comment (OlStani) [reply submissions/OlStaniev/DOM/script/script.js:8]: щоб бургер меню закривалося по кліку в будь-якому місці

File: submissions/OlStaniev/DOM/script/script.js (hunk 1 above)
Comment (OlStani) [reply submissions/OlStaniev/DOM/script/script.js:45]: > I mean write a function in which we call function, declare handlers, and others. For example:
> 
> ```js
> const init = () => {
>    .....
>    renderMenuList();
>    menu.addEventListener('click', getEpisode);
>    .....
> }
> ```

як будете мати час напишіть будь-ласка, бо трохи не розумію https://t.me/olstaniev

File: submissions/OlStaniev/DOM/script/script.js [hunk 8]
```diff
@@ -0,0 +1,46 @@
+import episodes from './text-episode.js';
+
+const menuList = document.querySelector('.nav__list');
+const introText = document.querySelector('.introText')
+const burgerIcon = document.querySelector('.header__burgerMenu')
```
Comment (madmaxWMFU) [inline submissions/OlStaniev/DOM/script/script.js:5]: Check your variables of naming.
Maybe the name burgerBtn will be correct. Because we have a button, not an icon.

File: submissions/OlStaniev/DOM/script/script.js [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+    if (!event.target.closest('.header__burgerMenu')) {
+        menuList.classList.remove('nav__listOpen')
+        burgerIcon.classList.remove('burgerCross')
+    }
+}
+
+function generateText(episodeId) {
+    const currentEpisode = episodes.find((episodeNumber) => episodeId === episodeNumber.id)
+    const storyText = `<div class="introText_wrapper"><h2>${currentEpisode.title}</h2><h3>${currentEpisode.subTitle}</h3><p>${currentEpisode.p1}</p><p>${currentEpisode.p2}</p><p>${currentEpisode.p3}</p></div>`
+    return storyText
```
Comment (madmaxWMFU) [inline submissions/OlStaniev/DOM/script/script.js:30]: Also, use destructuring.
Write code more readable, not one line.
Get rid of the extra variable, just return the template.

File: submissions/OlStaniev/DOM/index.html [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+        <div class="introText">
+            <div class="introText__wrapper">
+            </div>
+        </div>
+    </main>
+</body>
+
+<script src="./script/script.js" type="module"></script>
+<script src="./script/text-episode.js" type="module"></script>
+
```
Comment (madmaxWMFU) [inline submissions/OlStaniev/DOM/index.html:34]: Check the page in the [validator](https://validator.w3.org/).
A tag `<script>` must be into the tag `<body>`.

**Review by madmaxWMFU — CHANGES_REQUESTED** [review]
Don't forget about the comma separated.
Do separate handler and function.
In future try to use [prettier](https://prettier.io/).

File: submissions/OlStaniev/DOM/script/script.js (hunk 5 above)
Comment (madmaxWMFU) [reply submissions/OlStaniev/DOM/script/script.js:8]: This is not good. Still, this menu should be opened/closed by pressing a button. At least assign to a tag `<body>`.

File: submissions/OlStaniev/DOM/script/text-episode.js [hunk 11]
```diff
@@ -0,0 +1,135 @@
+const episodes = [{
+    id:'ep1',
+    title:`Episode I`,
+    subTitle:`THE PHANTOM MENACE`,
+    p1:`Turmoil has engulfed the
+    Galactic Republic. The taxation
+    of trade routes to outlying star
+    systems is in dispute.`,
+    p2:`Hoping to resolve the matter
```
Comment (madmaxWMFU) [inline submissions/OlStaniev/DOM/script/text-episode.js:9]: If the next episode has description in 10 paragraphs?
Maybe create property description which have array of paragraphed.

**Review by madmaxWMFU — CHANGES_REQUESTED** [review]
You did not change some recommendation. Do it and approving are your.

**Comment by madmaxWMFU** [comment]
Well done)
